### ISSUE TITLE
[BUG] LinearModel silently ignores penalty='elasticnet' parameter due to incorrect filtering

### ISSUE THREAD

**Issue by barstow2** [awslabs/autogluon#5175]
**Description**
AutoGluon's `LinearModel` silently ignores the `penalty='elasticnet'` parameter for classification tasks, preventing users from using elastic net regularization. The model appears to train successfully but uses L2 regularization instead.

**Expected behavior**
When specifying `penalty='elasticnet'` in LinearModel hyperparameters, the model should use elastic net regularization with the specified `l1_ratio`.

**Actual Behavior**
- AutoGluon accepts and reports `penalty='elasticnet'` in `predictor.info()`
- But the underlying sklearn LogisticRegression uses `penalty='l2'`
- The elastic net parameter is silently filtered out with no error message
- Users get L2 regularization instead of the requested elastic net

**To Reproduce**
```
from autogluon.tabular import TabularPredictor
import pandas as pd

# Create sample data
data = pd.DataFrame({
    'feature1': [1, 2, 3, 4, 5],
    'feature2': [2, 4, 6, 8, 10], 
    'target': [0, 1, 0, 1, 0]
})

# This should use elastic net but doesn't
predictor = TabularPredictor(label="target", problem_type="binary").fit(
    train_data=data,
    hyperparameters={
        "LR": {
            'penalty': 'elasticnet',
            'solver': 'saga',
            'l1_ratio': 0.5
        }
    }
)

# Note that AutoGluon reports 'elasticnet', but sklearn has 'l2'
sklearn_model = predictor._trainer.load_model("LinearModel").model
print(f"AutoGluon reports: {predictor.info()['model_info']['LinearModel']['hyperparameters']['penalty']}")
print(f"sklearn actually has: {sklearn_model.penalty}")



# Workaround that I used to test
from autogluon.tabular.models.lr.hyperparameters import parameters
parameters.preprocess_params_set.remove("penalty")
```

**Proposed Fix**
Remove "penalty" from `previous_params_set` in `parameters.py`:

```
preprocess_params_set = {
    "vectorizer_dict_size",
    "proc.ngram_range",
    "proc.skew_threshold", 
    "proc.impute_strategy",
    # Remove "penalty" from here
    "handle_text",
}
```

I'm happy to create a PR, this seems like a really simple fix.

**Installed Versions**

<details>
INSTALLED VERSIONS
------------------
date                   : 2025-06-26
time                   : 16:35:50.816039
python                 : 3.10.16.final.0
OS                     : Windows
OS-release             : 10
Version                : 10.0.22631
machine                : AMD64
processor              : Intel64 Family 6 Model 183 Stepping 1, GenuineIntel
num_cores              : 32
cpu_ram_mb             : 32424.0703125
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : [6523]
avail_disk_size_mb     : None

accelerate             : 1.5.2
autogluon              : 1.2
autogluon.common       : 1.2
autogluon.core         : 1.2
autogluon.features     : 1.2
autogluon.multimodal   : 1.2
autogluon.tabular      : 1.2
autogluon.timeseries   : 1.2
boto3                  : 1.38.0
catboost               : 1.2.7
coreforecast           : 0.0.16
defusedxml             : 0.7.1
einops                 : None
evaluate               : 0.4.3
fastai                 : 2.7.19
fugue                  : 0.9.1
gluonts                : 0.16.1
huggingface_hub        : 0.30.2
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.6
joblib                 : 1.4.2
jsonschema             : 4.23.0
lightgbm               : 4.4.0
lightning              : 2.5.1
matplotlib             : 3.9.4
mlforecast             : 0.13.6
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.9.1
numpy                  : 1.26.4
nvidia-ml-py3          : None
omegaconf              : 2.3.0
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.7
optimum                : None
optimum-intel          : None
orjson                 : 3.10.16
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 11.1.0
psutil                 : 6.1.1
pyarrow                : 19.0.0
pytesseract            : 0.3.10
pytorch-metric-learning: 2.3.0
pytorch_lightning      : 2.5.1
ray                    : 2.37.0
requests               : 2.32.3
scikit-image           : 0.24.0
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.13.1
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.8.5
statsforecast          : 1.7.8
tabpfn                 : None
tensorboard            : 2.19.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.5.1
torchmetrics           : 1.2.1
torchvision            : 0.15.2a0
tqdm                   : 4.67.1
transformers           : 4.51.3
utilsforecast          : 0.2.3
vowpalwabbit           : None
xgboost                : 2.1.4
</details>

**Comment by munro** [comment]
I think it's much worse than that, also ignores `L1` / `L2` for classification

### Here we get the model_cls from `penalty`
https://github.com/autogluon/autogluon/blob/d107dc495b90132fc6a6674f50614ecc9aec83eb/tabular/src/autogluon/tabular/models/lr/lr_model.py#L52-L53
https://github.com/autogluon/autogluon/blob/d107dc495b90132fc6a6674f50614ecc9aec83eb/tabular/src/autogluon/tabular/models/lr/lr_model.py#L71-L79

For classification it uses simply `sklearn.linear_model.LogisticRegression`
https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html

### Then later we get the params and init the model_cls
https://github.com/autogluon/autogluon/blob/d107dc495b90132fc6a6674f50614ecc9aec83eb/tabular/src/autogluon/tabular/models/lr/lr_model.py#L177
https://github.com/autogluon/autogluon/blob/d107dc495b90132fc6a6674f50614ecc9aec83eb/tabular/src/autogluon/tabular/models/lr/lr_model.py#L221

### But if you follow the code to `preprocess_params_set`, you see we're always excluding passing in `penalty`
https://github.com/autogluon/autogluon/blob/d107dc495b90132fc6a6674f50614ecc9aec83eb/tabular/src/autogluon/tabular/models/lr/hyperparameters/parameters.py#L16

### So thus `elasticnet` or any of the other penalties never make it in to `model_cls(**params)`

-----

i see op suggested just removing from `preprocess_params_set` but that would break the regression when it switches models
